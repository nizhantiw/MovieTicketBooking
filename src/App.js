import './App.css';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import Header from 'components/Header/Header';
import PageNotFound from 'containers/shared/PageNotFound/PageNotFound';
import { adminRoutes, clientRoutes } from 'routes';
import Home from 'containers/client/Home/Home';
import LoginClick from './containers/client/Login/LoginClick'
import RegisterClick from './containers/client/Login/RegisterClick'

function App() {
  const renderRoutes = routes => {
    return routes.map(route => {
      const { path, component, exact } = route;
      return (
        <Route 
          key={path}
          path={path} 
          component={component} 
          exact={exact} 
        />
      );
    });
  };

  return (
    <div className="App">
      <Router>
        <Header />
        <Switch>
        <Route exact path="/" component={Home} />
        <Route exact path="/login/login" component={LoginClick} />
        <Route exact path="/login/register" component={RegisterClick} />
          {renderRoutes(clientRoutes)}
          {renderRoutes(adminRoutes)} 
          <Route path="*" component={PageNotFound} />
        </Switch>
      </Router>
    </div>
  );
}

export default App;
