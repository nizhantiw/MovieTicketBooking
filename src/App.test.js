import App from './App';
import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

configure({adapter: new Adapter()});

it("renders correctly", () => {
  const wrapper = shallow(
    <App />
  );
  expect(wrapper).toMatchSnapshot();
});

it('renders the component', () => {
  const wrapper = shallow(<App />);
  expect(wrapper.exists()).toBe(true);
});


it('should show the text', () => {
    const toggleInstance = shallow(<App />);
    const element = toggleInstance.find('div');
    expect(element.text()).toMatchSnapshot();
   });