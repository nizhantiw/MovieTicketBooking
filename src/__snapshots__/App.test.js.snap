// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `ShallowWrapper {}`;

exports[`should show the text 1`] = `"<BrowserRouter />"`;
